### PR TITLE
use contentprovider API to load readonly test report

### PR DESCRIPTION
**Pull request by ansyral**
Signed-off-by: xuzho <xuzho@microsoft.com>

no need to save/not save test report any more.

### CODE REVIEW

File: extension/src/extension.ts [hunk 1]
```diff
@@ -22,12 +22,13 @@ import * as vscode from 'vscode';
...
 import { JUnitCodeLensProvider } from './junitCodeLensProvider';
 import { TestResourceManager } from './testResourceManager';
-import { OutputChannel, SnippetString, ExtensionContext, window } from 'vscode';
+import { OutputChannel, ExtensionContext, Uri } from 'vscode';
 import { TestSuite, TestLevel } from './protocols';
 import { ClassPathManager } from './classPathManager';
 import { TestResultAnalyzer } from './testResultAnalyzer';
 import TelemetryReporter from 'vscode-extension-telemetry';
 import { Logger, LogLevel } from './logger';
+import TestReportProvider, { encodeTestSuite } from './testReportProvider';
```
Comment (yaohaizh) [inline extension/src/extension.ts]: Should avoid use the mixed import style.

**Comment by ansyral** [comment]
#1 #7 #23

File: extension/src/extension.ts [hunk 2]
```diff
@@ -22,12 +22,13 @@ import * as vscode from 'vscode';
 import { Commands, Configs, Constants } from './commands';
 import { JUnitCodeLensProvider } from './junitCodeLensProvider';
 import { TestResourceManager } from './testResourceManager';
-import { OutputChannel, SnippetString, ExtensionContext, window } from 'vscode';
+import { OutputChannel, ExtensionContext, Uri } from 'vscode';
```
Comment (yaohaizh) [inline extension/src/extension.ts]: Already import the vscode as *.  This import is not necessary.

File: extension/src/extension.ts [hunk 3]
```diff
@@ -218,31 +220,12 @@ async function runSingleton(javaHome: string, tests: TestSuite[] | TestSuite, st
...
 function showDetails(test: TestSuite) {
     const editor = vscode.window.activeTextEditor;
-    const dir = path.dirname(editor.document.uri.fsPath);
-    const uri = vscode.Uri.parse(`${Constants.TEST_OUTPUT_SCHEME}:${path.join(dir, 'test-result-' + test.test)}`);
+    const uri: Uri = encodeTestSuite(editor.document.uri, test);
     return vscode.workspace.openTextDocument(uri).then(doc => {
-        return vscode.window.showTextDocument(doc, editor.viewColumn + 1).then(edit => {
-            edit.insertSnippet(new SnippetString(getTestReport(test)), new vscode.Position(0, 0));
-        });
+        return vscode.window.showTextDocument(doc, editor.viewColumn + 1);
```
Comment (yaohaizh) [inline extension/src/extension.ts]: Suggestion, simple sentence  can remove return and {}

File: extension/src/testReportProvider.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
+import { TestResourceManager } from "./testResourceManager";
+import { TestSuite, TestLevel } from "./protocols";
+import { TextDocumentContentProvider, Uri } from "vscode";
+
+import * as os from 'os';
```
Comment (yaohaizh) [inline extension/src/testReportProvider.ts]: Should add eslint files for checking styles

File: extension/src/testReportProvider.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    }
+}
+
+export function encodeTestSuite(uri: Uri, test: TestSuite): Uri {
+    const query = JSON.stringify([uri.toString(), test.test.replace('#', '%23')]);
+    return Uri.parse(`${TestReportProvider.scheme}:test-report?${query}`);
+}
+
+export function decodeTestSuite(uri: Uri): [Uri, string] {
+    let [target, test] = <[string, string]>JSON.parse(uri.query);
```
Comment (yaohaizh) [inline extension/src/testReportProvider.ts]: const

File: extension/src/extension.ts (hunk 2 above)
Comment (ansyral) [reply extension/src/extension.ts]: I added this to not introduce `vscode.` frequently. Removing `windows` as it is lowercase and might be confused with local variables. I think it is good to keep consistent. Which way do you think is better? `import * as vscode from vscode` or import all the used type in vscode module explicitly

Comment (yaohaizh) [reply extension/src/extension.ts]: Either import as * or import separate. Either way works but not mixed.

File: extension/src/testReportProvider.ts (hunk 4 above)
Comment (ansyral) [reply extension/src/testReportProvider.ts]: already added tslint.json at the end of last release, however I didn't make it too strict as it introduced many errors. might need to open a separate pr to update tslint.json and fix all style errors.

File: extension/src/testReportProvider.ts [hunk 6]
```diff
@@ -0,0 +1,57 @@
+import { TestLevel, TestSuite } from "./protocols";
+import { TestResourceManager } from "./testResourceManager";
+
+import * as os from 'os';
+import { TextDocumentContentProvider, Uri } from "vscode";
+
+export class TestReportProvider implements TextDocumentContentProvider {
+
+    static scheme = 'test-report';
```
Comment (yaohaizh) [inline extension/src/testReportProvider.ts]: Const?

Comment (ansyral) [reply extension/src/testReportProvider.ts]: const isn't allowed as class member

**Pull request merged**